src/figment.rs

```diff
@@ -273,6 +303,13 @@ impl Figment {
         self.provide(provider, Order::Merge)
     }
 
+    /// Merges `provider` into the current figment while concatenating vectors.
```

Comment:
Same thing here. We should have a minimal but representative example, and then delegate the rest to the link.

Reply:
Fair point, I added documentation to each individual strategy function in 28c86af.